tests/infra/signing.py

```diff
@@ -1,121 +1,12 @@
-    :return: an initialized EC key
-    """
-    if hasattr(ext_key, "private_numbers"):
-        priv_nums = ext_key.private_numbers()
-        pub_nums = priv_nums.public_numbers
-    else:
-        priv_nums = None
-        pub_nums = ext_key.public_numbers()
-
-    if pub_nums.curve.name == "secp256r1":
-        curve = P256
-    elif pub_nums.curve.name == "secp384r1":
-        curve = P384
-    elif pub_nums.curve.name == "secp521r1":
-        curve = P521
-    else:
-        raise NotImplementedError("unsupported curve")
-
-    cose_key = {}
-    if pub_nums:
-        cose_key.update(
-            {
-                EC2KpCurve: curve,
-                EC2KpX: pub_nums.x.to_bytes(curve.size, "big"),
-                EC2KpY: pub_nums.y.to_bytes(curve.size, "big"),
-            }
-        )
-    if priv_nums:
-        cose_key.update(
-            {
-                EC2KpD: priv_nums.private_value.to_bytes(curve.size, "big"),
-            }
-        )
-    return EC2Key.from_dict(cose_key)
-
-
-def default_algorithm_for_key(key) -> str:
-    """
-    Get the default algorithm for a given key, based on its
-    type and parameters.
-    """
-    if isinstance(key, EllipticCurvePublicKey):
-        if isinstance(key.curve, ec.SECP256R1):
-            return "ES256"
-        elif isinstance(key.curve, ec.SECP384R1):
-            return "ES384"
-        elif isinstance(key.curve, ec.SECP521R1):
-            return "ES512"
-        else:
-            raise NotImplementedError("unsupported curve")
-    else:
-        raise NotImplementedError("unsupported key type")
-
-
-def get_priv_key_type(priv_pem: str) -> str:
-    key = load_pem_private_key(priv_pem.encode("ascii"), None, default_backend())
-    if isinstance(key, EllipticCurvePrivateKey):
-        return "ec"
-    raise NotImplementedError("unsupported key type")
-
-
-def cert_fingerprint(cert_pem: Pem):
-    cert = load_pem_x509_certificate(cert_pem.encode("ascii"), default_backend())
-    return cert.fingerprint(hashes.SHA256()).hex().encode("utf-8")
-
-
-def create_cose_sign1(
-    payload: bytes,
-    key_priv_pem: Pem,
-    cert_pem: Pem,
-    additional_headers: Optional[dict] = None,
-) -> bytes:
-    key_type = get_priv_key_type(key_priv_pem)
-
-    cert = load_pem_x509_certificate(cert_pem.encode("ascii"), default_backend())
-    alg = default_algorithm_for_key(cert.public_key())
-    kid = cert_fingerprint(cert_pem)
-
-    headers = {cose.headers.Algorithm: alg, cose.headers.KID: kid}
-    headers.update(additional_headers or {})
-    msg = Sign1Message(phdr=headers, payload=payload)
-
-    key = load_pem_private_key(key_priv_pem.encode("ascii"), None, default_backend())
-    if key_type == "ec":
-        cose_key = from_cryptography_eckey_obj(key)
-    else:
-        raise NotImplementedError("unsupported key type")
-    msg.key = cose_key
-
-    return msg.encode()
+from pycose.messages import Sign1Message  # type: ignore
 
 
 def get_cert_key_type(cert_pem: str) -> str:
```

Comment:
What is the reason to leave some functions here?

Reply:
They're not needed for the tool or for signing, they're only used for tests, so I'd rather not release them.

Reply:
It looks like actually they are not used now. So if we don't have a plan to use it again in near future, it might be better to remove the file.

Reply:
I think that's a mistake on my part, the governance audit test should require them.

Reply:
Done now.